core/new-gui/src/app/common/type/workflow.ts

```diff
@@ -1,12 +1,38 @@
-export interface Workflow extends Readonly<{
+import { Breakpoint, OperatorLink, OperatorPredicate, Point } from '../../workspace/types/workflow-common.interface';
+
+/**
+ * CachedWorkflow is used to store the information of the workflow
+ *  1. all existing operators and their properties
+ *  2. operator's position on the JointJS paper
+ *  3. operator link predicates
+ *
+ * When the user refreshes the browser, the CachedWorkflow interface will be
+ *  automatically cached and loaded once the refresh completes. This information
+ *  will then be used to reload the entire workflow.
+ *
+ */
+export interface WorkflowInfo {
+  operators: OperatorPredicate[];
+  operatorPositions: { [key: string]: Point };
+  links: OperatorLink[];
+  breakpoints: Record<string, Breakpoint>;
+}
+
+export interface Workflow {
```

Comment:
seems we should swap the names of these two interfaces, this interface is more about "info" (creation time, modified time, etc.), and the above interface is the actual workflowinte****

Reply:
It's a design choice. `Workflow` name is mapped throughout all three tiers including the database. I don't want to change it. `WorkflowInfo` was renamed from `CachedWorkflow`. I find `CachedWorkflow` confusing with `Workflow`. Maybe we need another name for `WorkflowInfo`: in the database, it is called `content` and previously called `workflowBody` which I also don't like either.